controllers/rollingupgrade_controller.go

```diff
@@ -875,3 +908,35 @@ func (r *RollingUpgradeReconciler) UpdateInstance(ctx *context.Context,
 	ch <- nil
 	return
 }
+
+func requiresRefresh(ec2Instance *autoscaling.Instance, definition *launchDefinition) bool {
+	if definition.kind == launchKindLaunchConfiguration {
+		if ec2Instance.LaunchConfigurationName != nil && *ec2Instance.LaunchConfigurationName != *definition.launchConfigurationName {
+			return true
+		}
+	} else if definition.kind == launchKindLaunchTemplate {
+		if ec2Instance.LaunchTemplate != nil && definition.launchTemplate != nil {
```

Comment:
Could we simplify this logic? we should think whether all these conditions are really required for us to know a launchtemplate is defined.
Is it not enough to simply say `else if aws.StringValue(ec2Instance.LaunchTemplate) != nil { ... }`? that way we can get rid of all the code below.

Reply:
LaunchTemplate is an object that have 3 fields - `LaunchTemplateName`, `LaunchTemplateId` and `Version`. So, need to check for the difference. Also, I'm testing what happens if LaunchTemplate refers to version via `$Default` and `$Latest` - additional checks may be required to determine if `$Latest` was updated.

Reply:
With LaunchConfigurations, when it drifts, the instances have the field `LaunchConfigurationName` as nil, so we can simply rely on that field being nil.
Is this not the same case with `LaunchTemplate`? when there is a drift, they should strip out that field and you can rely on it in the same manner.